File: 448-find-all-numbers-disappeared-in-an-array/solution.ts
function findDisappearedNumbers(nums: number[]): number[] {
  const n = nums.length;
  const result = {};
  for (let i = 1; i <= n; i++) result[i] = 1;
  for (let i = 0; i < n; i++) delete result[nums[i]];
  return Object.keys(result).map((i) => parseInt(i));
}

File: 448-find-all-numbers-disappeared-in-an-array/solution.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n + 1);
        for(int &x : ans) x = 1;
        for(int &x : nums) ans[x] = 0;
        for(int i = 0, j = 1; j <= n; j++) {
            if(ans[j]) ans[i++] = j;
            if(j == n) {
                for(int k = n; k >= i; k--) ans.erase(ans.begin() + k);
            }
        }
        return ans;
    }
};
